paperqa/agents/tools.py

```diff
@@ -76,7 +69,11 @@ def status(self) -> str:
 
     def record_action(self, action: ToolRequestMessage) -> None:
         self.session.add_tokens(action)
-        self.tool_history.append([tc.function.name for tc in action.tool_calls])
+        self.session.tool_history.append([tc.function.name for tc in action.tool_calls])
+
+    def query_tool_history(self, tool_name: str) -> list[int]:
+        """Return the step indices of the tool calls that used the given tool."""
+        return [i for i, tc in enumerate(self.session.tool_history) if tool_name in tc]
```

Comment:
```suggestion
        return [i for i, tnames in enumerate(self.session.tool_history) if tool_name in tnames]
```

Let's use a name besides `tc` because that implies `ToolCall`